jhub_apps/spawner/spawner_creation.py

```diff
@@ -56,6 +56,16 @@ def get_args(self):
                     command = Command(args=GENERIC_ARGS + custom_cmd.split())
                 else:
                     command: Command = COMMANDS.get(framework)
+
+                repository = self.user_options.get("repository")
+                if repository:
+                    logger.info(f"repository found: {repository}")
+                    command.args.extend([
+                        f"--repo={repository.get('url')}",
+                        f"--repofolder=/tmp/{self.name}",
+                        f"--repobranch={repository.get('ref')}"
+                    ])
```

Comment:
This will presumably clone the repository each time as the specified folder is in `/tmp`.
- a) is the `name` guaranteed to be unique?
- b) should it reuse the existing repository if it exists on the user's disk and has a matching `ref`?
   - I worry that if a user deploys a repository from git and then clones the same repository on disk, modifies the path and sees no change on redeployment they may think it is broken, but in fact only the committed and pushed version is expected to be used.
   - it may be ok on the backend side as-is, but if we do want it this way it needs a lot of attention to how the user is informed about it on the frontend
- c) what happens when redeploying if the repo already exists in `/tmp`? By skimming the code of `jhsingle-native-proxy` I think it will not update it, so if user initialized the repo from `main`, committed, pushed and redeployed then it _might_  be using the old version? I am not sure because this comes down to how `/tmp` is mounted and whether redeployment always means we get a fresh `/tmp` (which is likely).

Reply:
>is the name guaranteed to be unique?

name is unique, but it will be a problem when editing the app (re-deployment locally), hence I have just added a unique suffix.

>b) should it reuse the existing repository if it exists on the user's disk and has a matching ref?

This is beyond the scope of the original plan and yes requires a lot of thought before considering add support. Although I agree, we need to make sure this is explicit in the UI messaging.

> what happens when redeploying if the repo already exists in /tmp? By skimming the code of jhsingle-native-proxy I think it will not update it, so if user initialized the repo from main, committed, pushed and redeployed then it might be using the old version? I am not sure because this comes down to how /tmp is mounted and whether redeployment always means we get a fresh /tmp (which is likely).

`/tmp` is fresh in in case of Nebari but for local deployment it won't be that's why I have just added a suffix in the name.